### PR TITLE
Support for core-cycle counts.

### PR DESCRIPTION
**Pull request by vext01**
These changes are paired with the core-cycles-msr PR on krun.

(It also includes the V8 fix @cfbolz encountered)

### CODE REVIEW

**Comment by vext01** [comment]
This is now synced with @snim2's changes on master.

File: build.sh [hunk 1]
```diff
@@ -62,11 +62,23 @@ echo "===> Working in $wrkdir"
 PATCH_DIR=`pwd`/patches/
 
 # XXX when we stabilise, fix the krun revision.
-fetch_krun() {
+build_krun() {
```
Comment (ltratt) [inline build.sh]: Should this be something like `build_initial_krun`?

File: patches/jruby.diff [hunk 2]
```diff
@@ -51,66 +51,73 @@ index 50a155d..0234ef7 100644
...
+@@ -1985,4 +1986,56 @@ public abstract class KernelNodes {
  
      }
  
-+    @CoreMethod(names = "read_ts_reg_start", needsSelf=false)
-+    public abstract static class ReadTSRegStartNode extends CoreMethodArrayArgumentsNode {
++    @CoreMethod(names = "read_core_cycles", needsSelf=false)
++    public abstract static class ReadCoreCyclesNode extends CoreMethodArrayArgumentsNode {
 +
 +        private static final BigInteger negAdjust = BigInteger.ONE.shiftLeft(64); // 2^64
```
Comment (ltratt) [inline patches/jruby.diff:61]: Did the indentation level just jump to 8?

**Comment by ltratt** [comment]
Looks pretty good to me (apart possibly from wonky indentation in the Java part?).

File: patches/jruby.diff (hunk 2 above)
Comment (vext01) [reply patches/jruby.diff:61]: Note that this is the first line of a new block. Looks OK to me, unless I am mistaken.

**Comment by vext01** [comment]
Addressed those bits.

File: patches/jruby.diff (hunk 2 above)
Comment (ltratt) [reply patches/jruby.diff:61]: Nope, you're right (oops!).

**Comment by ltratt** [comment]
OK, thanks. Let's wait for @cfbolz to do a pass.

**Comment by vext01** [comment]
and can we merge this and the krun change at the same time too?

**Comment by ltratt** [comment]
Yes, although I don't think we need to be super-worried about pressing the merge buttons completely simultaneously :)

**Comment by vext01** [comment]
I think these two PRs are good to merge.

**Comment by ltratt** [comment]
Still waiting for @cfbolz's OK on this PR.

**Comment by cfbolz** [comment]
Sorry for the delay, this looks good to me.

**Comment by vext01** [comment]
OK to squash?

**Comment by ltratt** [comment]
Yep, let's squash then I can merge.

**Comment by vext01** [comment]
splat

**Comment by ltratt** [comment]
Can we fix the typo in the "fix import path" commit? [Assuming it needs to be a separate commit.]

**Comment by vext01** [comment]
splatted into one

**Pull request merged**